src/realm/cluster_tree.cpp

```diff
@@ -892,17 +957,26 @@ void ClusterTree::insert_fast(ObjKey k, const FieldValues& init_values, ClusterN
     m_size++;
 }
 
-ClusterNode::State ClusterTree::insert(ObjKey k, const FieldValues& init_values)
+Obj ClusterTree::insert(ObjKey k, const FieldValues& init_values)
```

Comment:
This is a little counterintuitive, why do we return the Obj just inserted, or we construct one to signal that we have inserted? Do we need some reference to the object right after the insertion?

Reply:
This is used when we create objects. So the goal is actually to return an object just created.